evalml/pipelines/components/estimators/regressors/catboost_regressor.py

```diff
@@ -57,9 +58,11 @@ def __init__(
         cb_parameters = copy.copy(parameters)
         if bootstrap_type is None:
             cb_parameters.pop("bootstrap_type")
+        cb_parameters["thread_count"] = n_jobs
```

Comment:
Regarding your UX question, it might be good to let users know via a warning that the `thread_count` parameter is ignored if passed in.

We used to do something similar for the `loss` parameter of the old elastic net parameter.

https://github.com/alteryx/evalml/blob/v0.25.0/evalml/pipelines/components/estimators/classifiers/elasticnet_classifier.py#L32

Reply:
I love this suggestion, added!!